src/toil/job.py

```diff
@@ -779,14 +778,19 @@ def makeString(x: Union[str, bytes, None]) -> str:
         # ID of this job description in the JobStore.
         self.jobStoreID: Union[str, TemporaryID] = TemporaryID()
 
-        # Mostly fake, not-really-executable command string that encodes how to
-        # find the Job body data that this JobDescription describes, and the
-        # module(s) needed to unpickle it.
+        # Actual command string we intend for the batch system to run somewhere
+        # Used internally by the leader.
+        # TODO: Really internal leader state!
+        self._worker_command: Optional[str] = None
+
+        # String that encodes how to find the Job body data that this
+        # JobDescription describes, and the module(s) needed to unpickle it.
+        # None if no body needs to run.
         #
-        # Gets replaced with/rewritten into the real, executable command when
-        # the leader passes the description off to the batch system to be
-        # executed.
-        self.command: Optional[str] = command
+        # Used to be the "command" until it was extended beyond making sense as
+        # an actual command, so now it's just a string with several
+        # space-separated fields. See attach_body() and get_body().
+        self._body_spec: Optional[str] = None
```

Comment:
Suggestion: Change to `self._body` instead of `self._body_spec` (so as to be in line with `attach_body()` and `get_body()`).

Also, not necessary, but I feel like it might be clearer (and self documenting) if this could either be separate `self` variables like `self.filestore_id` and `self.command` OR a json like `{"filestore_id": 1234, "command": ["echo", "echo"]}`.  That way I wouldn't have to scroll to `attach_body()` and `get_body()` to intuit what the contents of `self._body_spec` might be.